lib/cocoapods/installer/sandbox_dir_cleaner.rb

```diff
@@ -0,0 +1,98 @@
+module Pod
+  class Installer
+    # Cleans up the sandbox directory by removing stale target support files and headers.
+    #
+    class SandboxDirCleaner
+      # @return [Sandbox] The sandbox directory that will be cleaned.
+      #
+      attr_reader :sandbox
+
+      # @return [Array<PodTarget>]
+      #         The list of all pod targets that will be installed into the Sandbox.
```

Comment:
is this correct or copy pasta?

Reply:
Correct in that I wrote it :) The list of pod targets that will be installed are used to compute which folders need to be deleted.